src/common/run_cmd.cc

```diff
@@ -78,3 +83,25 @@ std::string run_cmd(const char *cmd, ...)
   oss << "run_cmd(" << cmd << "): terminated by unknown mechanism";
   return oss.str();
 }
+#else
+std::string run_cmd(const char *cmd, ...)
+{
+  SubProcess p(cmd, SubProcess::CLOSE, SubProcess::PIPE, SubProcess::CLOSE);
+
+  va_list ap;
+  va_start(ap, cmd);
+  const char *c = cmd;
+  c = va_arg(ap, const char*);
+  while (c != NULL) {
+    p.add_cmd_arg(c);
+    c = va_arg(ap, const char*);
+  }
+  va_end(ap);
+
+  if (p.spawn() == 0) {
+    p.join();
+  }
+
+  return "";
```

Comment:
error handling?

Reply:
Right, all we have to do here is propagate the ``SubProcess`` error string.